client/src/app/core/ui-services/base-sort-list.service.ts

```diff
@@ -218,59 +218,72 @@ export abstract class BaseSortListService<V extends BaseViewModel> {
+                        return 1;
+                    } else if (!secondProperty) {
+                        return -1;
                     } else {
-                        if (
-                            (firstProperty === null || firstProperty === undefined) &&
-                            (secondProperty === null || secondProperty === undefined)
-                        ) {
-                            return 1;
-                        }
-                        switch (typeof firstProperty) {
-                            case 'boolean':
-                                if (firstProperty === false && secondProperty === true) {
-                                    return -1;
-                                } else {
-                                    return 1;
-                                }
-                            case 'number':
-                                return firstProperty > secondProperty ? 1 : -1;
-                            case 'string':
-                                if (!firstProperty) {
-                                    return 1;
-                                }
-                                return intl.compare(firstProperty, secondProperty);
-                            case 'function':
-                                const a = firstProperty();
-                                const b = secondProperty();
-                                return intl.compare(a, b);
-                            case 'object':
-                                if (firstProperty instanceof Date) {
-                                    return firstProperty > secondProperty ? 1 : -1;
-                                } else {
-                                    return intl.compare(firstProperty.toString(), secondProperty.toString());
-                                }
-                            case 'undefined':
-                                return 1;
-                            default:
+                        throw new TypeError('sorting of items failed because of mismatched types');
+                    }
+                } else {
+                    if (
+                        (firstProperty === null || firstProperty === undefined) &&
+                        (secondProperty === null || secondProperty === undefined)
+                    ) {
+                        return 1;
+                    }
+                    switch (typeof firstProperty) {
+                        case 'boolean':
+                            if (firstProperty === false && secondProperty === true) {
                                 return -1;
-                        }
+                            } else {
+                                return 1;
+                            }
+                        case 'number':
+                            return firstProperty > secondProperty ? 1 : -1;
+                        case 'string':
+                            return intl.compare(firstProperty, secondProperty);
```

Comment:
this was my idea before:
```js
if (!!firstProperty !== !!secondProperty) {
    return -1;
} else {
    return intl.compare(firstProperty, secondProperty);
}

All empty strings should be sorted at the bottom. Yet it can lead to unexpected sorting